sn_transfers/src/cashnotes/spend_reason.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use bls::{Ciphertext, PublicKey, SecretKey};
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use xor_name::XorName;
+
+use crate::{DerivationIndex, Hash, Result, TransferError};
+
+/// The attached metadata or reason for which a Spend was spent
+#[derive(Default, Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum SpendReason {
+    #[default]
+    None,
+    /// Reference to network data
+    NetworkData(XorName),
```

Comment:
better have this as NetworkAddress instead of XorName
so that a refer to Register will be more convenient.

Reply:
I agree and wanted to, but this creates a circular dependency... We should probably move NetworkAdress to another crate?